src/plugins/auto/plugin.cpp

```diff
@@ -345,10 +345,28 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
              strDevices += ((iter + 1) == supportDevices.end()) ? "" : ",";
              LOG_INFO("[AUTOPLUGIN]:device:%s, priority:%ld", iter->deviceName.c_str(), iter->devicePriority);
         }
-        // clone the network, in case of reshape conflict
-        CNNNetwork clonedNetwork = InferenceEngine::details::cloneNetwork(network);
+        CNNNetwork clonedNetwork;
+        std::string clonedModelPath = modelPath;
+        try {
+            // if network is valid
+            network.getFunction();
+            LOG_INFO("[AUTOPLUGIN]:load with CNN network");
+            // clone the network, in case of reshape conflict
+            clonedNetwork = InferenceEngine::details::cloneNetwork(network);
+        } catch (...) {
+            // model path, enable model load with single device situation
+            if (supportDevices.size() > 1) {
+                clonedNetwork = GetCore()->ReadNetwork(modelPath, std::string());
+                // do we really need to disable model path?
+                clonedModelPath = "";
+                LOG_INFO("[AUTOPLUGIN]:load with CNN network");
+            } else {
+                LOG_INFO("[AUTOPLUGIN]:load with model path");
```

Comment:
Could we check model_path just after FilterDevice?

> auto supportDevicesByConfig = FilterDevice(metaDevices, filterConfig);

Reply:
updated